### PR TITLE
App router redirectUrl for hash based router

### LINKED ISSUE #997: Invalid URL created when using Next.js 13 app directory (http://localhost:3000/sign-inredirect_url=http:/localhost:3000/protected)
- [X] Review the documentation: https://docs.clerk.com/
- [X] Search for existing issues: https://github.com/clerkinc/javascript/issues
- [X] Go through package changelog files.
- [X] Provide the Frontend API key from your application dashboard. (pk_test_cHJvdWQtZGFuZS04Ni5jbGVyay5hY2NvdW50cy5kZXYk)

## Package + Version

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [X] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`
- [ ] other:

### Dependencies + versions

**Pages directory repo that works**: https://github.com/r3Fuze/next-pages-dir-clerk-redirect
**App directory repo that doesn't work**: https://github.com/r3Fuze/next-app-dir-clerk-redirect

Dependencies for non-functional app directory version.

```json
"dependencies": {
  "@clerk/nextjs": "^4.11.14",
  "@types/node": "18.15.11",
  "@types/react": "18.0.31",
  "@types/react-dom": "18.0.11",
  "autoprefixer": "10.4.14",
  "eslint": "8.37.0",
  "eslint-config-next": "13.2.4",
  "next": "13.2.4",
  "postcss": "8.4.21",
  "react": "18.2.0",
  "react-dom": "18.2.0",
  "tailwindcss": "3.3.0",
  "typescript": "5.0.2"
}
```

### Browser/OS

Firefox  112, Node 19.7.0, Windows 10

## Description

Clerk will create an invalid URL when used with the Next.js 13 app directory. A (mostly) identical version using the pages directory works fine.

Both versions were created by following the steps in the beta docs ([pages](https://beta-docs.clerk.com/quickstarts/nextjs/stable), [app](https://beta-docs.clerk.com/quickstarts/nextjs/app-router))

See the `Dependencies + versions` section for repos for both versions.

#### Reproduction steps for the app directory version
- Be signed out on the home page
- Navigate to `/protected`
- Redirected to `http://localhost:3000/sign-in?redirect_url=http%3A%2F%2Flocalhost%3A3000%2Fprotected`
- Sign in with GitHub or email (I've only tried those two, but they both fail)
- Redirected to `http://localhost:3000/sign-inredirect_url=http:/localhost:3000/protected`

Notice the missing `?` between `sign-in` and `redirect_url`. There is also a missing `/` in the protocol for the `redirect_url`

The same steps in the pages directory version create a valid URL at the end and redirect to `/protected` without problems.

Related support thread from the Discord: https://discord.com/channels/856971667393609759/1070730863077236846

Let me know if you need any more information.

### PR DESCRIPTION
**Pull request by nikosdouvlis**
## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [x] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.

Currently, useRouter.queryParams returns the params the router parses. The hash and virtual routers only parse the params within the URL fragment, so any params appended to the normal URL by the user are ignored.

Before the change:
1. Path routing
A SignIn component mounted at https://example.com/sign-in?redirect_to=/protected will correctly redirect to /protected
2. Hash routing with hash param
A SignIn component mounted at https://example.com/sign-in#/?redirect_to=/protected will correctly redirect to /protected
3. Hash routing with normal param
A SignIn component mounted at https://example.com/sign-in?redirect_to=/protected will ignore the redirect_to hash param

Before the change:
1 + 2 remain the same
2. Hash routing with normal param
A SignIn component mounted at https://example.com/sign-in?redirect_to=/protected will read the normal param and redirect to /protected

If both hash and normal params exists, the hash params will have priority - no breaking changes here.

Regarding the nextjs change: concatenating pathname and searchParams resulted in broken URLs like:
http://localhost/sign-inredirectTo=/protected - notice the missing ?.

Since no changes are needed, we can simply reexport the components as well.

Closes #997

### CODE REVIEW

**Review by desiprisg — APPROVED** [review]

File: packages/clerk-js/src/ui/common/authPropHelpers.ts [hunk 1]
```diff
@@ -21,25 +21,24 @@ type ExtractAuthPropOptions =
...
-export const extractAuthProp = (
-  key: ExtractAuthUrlKey,
-  { ctx, queryParams, displayConfig }: ExtractAuthPropOptions,
-): string => {
-  const snakeCaseField = camelToSnake(key);
-  const queryParamValue = queryParams[snakeCaseField];
+export const getRedirectProp = (key: ExtractAuthUrlKey, opts: ExtractAuthPropOptions): string => {
+  const { ctx, displayConfig, queryParams: routerParams, currentUrl = new URL(window.location.href) } = opts;
+  const urlParams = currentUrl.searchParams;
+  const snakeCasedKey = camelToSnake(key);
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/common/authPropHelpers.ts:34]: 💯

**Review by SokratisVidros — APPROVED** [review]

File: packages/nextjs/src/app-beta/client/ui-components.tsx [hunk 2]
```diff
@@ -1,48 +1,11 @@
...
   OrganizationSwitcher,
-  SignIn as OgSignIn,
-  SignUp as OgSignUp,
-  UserButton as OgUserButton,
+  SignIn,
+  SignUp,
+  UserButton,
   UserProfile,
 } from '@clerk/clerk-react';
-import type { SignInProps, SignUpProps, UserButtonProps } from '@clerk/types';
```
Comment (SokratisVidros) [inline packages/nextjs/src/app-beta/client/ui-components.tsx:11]: 💯

File: packages/clerk-js/src/ui/common/authPropHelpers.ts [hunk 3]
```diff
@@ -21,25 +21,24 @@ type ExtractAuthPropOptions =
...
- * Querystring parameters > ClerkJS component props > Display configuration payload
+ * Querystring parameters from the router > current url params > ClerkJS component props > Display configuration payload
  */
-export const extractAuthProp = (
-  key: ExtractAuthUrlKey,
-  { ctx, queryParams, displayConfig }: ExtractAuthPropOptions,
-): string => {
-  const snakeCaseField = camelToSnake(key);
-  const queryParamValue = queryParams[snakeCaseField];
+export const getRedirectProp = (key: ExtractAuthUrlKey, opts: ExtractAuthPropOptions): string => {
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/common/authPropHelpers.ts:31]: @nikosdouvlis Let's unit tests as this function has become very critical for the redirection logic.

**Review by SokratisVidros — APPROVED** [review]

**Comment by nikosdouvlis** [comment]
We decided to revert the `extractAuthProp` change for now, and instead prompt the users to switch to path-based routing.

**Comment by SokratisVidros** [comment]
@nikosdouvlis Did we notice any issues with the change in extractAuthProp?

**Pull request merged**